index.js

```diff
@@ -2,33 +2,35 @@
 
 /* eslint-disable no-underscore-dangle */
 class NotificationBase {
-
     /**
      * Constructor for Build Notification
      * @method constructor
      * @param {Object}      config              Configuration specific to the plugin
-     * @param {Object}      server              API server
-     * @param {String}      eventName           Event name the notification plugin listen to
      * @return {NotificationBase}
      */
-    constructor(config, server, eventName) {
+    constructor(config) {
         this.config = config;
-        this.server = server;
-        this.eventName = eventName;
+    }
+
+    /**
+     * Get event list for receiving Screwdriver API events
+     * @method events
+     * @return {Array}
+     */
+    get events() {
+        return ['build_status'];
     }
 
     /**
      * Listen to the event and notify users based on their job settings
      * @method notify
-     * @return {Promise}
      */
-    notify() {
-        return this._notify();
+    notify(buildData) {
+        return this._notify(buildData);
```

Comment:
I think that logging an error is good idea, and it is the only thing we can do.
There is no way to recover or handle the error since it depends on an implementation and here is inside of an event handler.